### PR TITLE
Smite!

### PR DESCRIPTION
**Pull request by JesterX666**
Really basic SMITE command for Admins.
Allows to instantly gib a player.
Only works on players, not on all living beings.

### CODE REVIEW

**Comment by Bod9001** [comment]
oh no, the bug from foma's PR Is causing issues

**Comment by ThatDan123** [comment]
I would move this all over to use the AdminCommandsManager.cs scripting system, just to keep them all together.

Has all the net message stuff for you to use there, just remember to put the parameters of your method in the right order.

**Comment by ThatDan123** [comment]
Also add in the admin action message stuff, see the other methods in the admin commands script.

**Comment by JesterX666** [comment]
> I would move this all over to use the AdminCommandsManager.cs scripting system, just to keep them all together.
> 
> Has all the net message stuff for you to use there, just remember to put the parameters of your method in the right order.

Why do we have two places for admin commands?  I've decided to put smite at the same place as all the other commands that targets a player (ban, kick, job ban, respawn, boink).

So, the "official" way to go is AdminCommandManager now?

**Comment by ThatDan123** [comment]
The bans and stuff are in the playerlist as theyre loaded in there. But yeah stick to the manager now.

**Comment by JesterX666** [comment]
Personally, I'm not a fan of he "versiontwo", "versionthree", "versionfour" kinda stuff.  We are still "alpha" and already have 5 ways to send the messages.  In the future, we might arrive to a point that we will have so many different type of calls, that it will become a mess.  

Reflection is known to be slow
Magic strings (i.e.: "string Action") is prone to error.  
Untyped parameters all in the same parameter variable is also prone to errors.

I can see what you tried to do there, it's a very noble idea to standardize stuff.  It's not "bad" code, as it works, but it's not the type of code I prefer.  But, for now, it works perfectly fine.  So, I'll use it.

**Comment by JesterX666** [comment]
@ThatDan123 changes made.  You can proceed for a second review.

File: UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs [hunk 1]
```diff
@@ -298,6 +294,36 @@ public void CmdSendBlockShuttleRecall(string adminId, string adminToken, bool to
...
+				return;
+
+			var players = PlayerList.Instance.GetAllByUserID(userToSmite);
+			if (players.Count != 0)
+			{
+				foreach (ConnectedPlayer player in players)
+				{
+					var message = $"{player.Name} has been smiten by {PlayerList.Instance.GetByUserID(adminId).Username}: Username: {player.Username}";
+					Logger.Log(message);
+
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: Add this here too

`UIManager.Instance.adminChatWindows.adminToAdminChat.ServerAddChatRecord(msg, null);
DiscordWebhookMessage.Instance.AddWebHookMessageToQueue(DiscordWebhookURLs.DiscordWebhookAdminLogURL, msg, "");`

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: And could you reformat the message to like this one:

$"{PlayerList.Instance.GetByUserID(adminId).Username}: {state} shuttle recalling.";

Admin username, then action

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Managers/AdminCommandsManager.cs]: `var message = $"{PlayerList.Instance.GetByUserID(adminId).Username}: Smitted Username: {player.Username}, {player.Name}";`

Something like that maybe, though whether its smited or smitted, stupid words :p

**Comment by JesterX666** [comment]
Done and tested

image

**Pull request merged**